poetry/installation/authenticator.py

```diff
@@ -61,8 +67,16 @@ def request(self, method: str, url: str, **kwargs: Any) -> requests.Response:
 
         proxies = kwargs.get("proxies", {})
         stream = kwargs.get("stream")
-        verify = kwargs.get("verify")
-        cert = kwargs.get("cert")
+
+        certs = self.get_certs_for_url(url)
+        verify = kwargs.get("verify", certs.get("verify"))
+        cert = kwargs.get("cert", certs.get("cert"))
+
+        if cert is not None:
+            cert = str(cert)
+
+        if isinstance(verify, pathlib.PurePath):
```

Comment:
```suggestion
        if verify is not None:
```

Same as the change @abn suggested for the block above.